slam/src/slam/fast_slam2/FastSLAM1Particle.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import scipy.stats
+from scipy.linalg import sqrtm
+from collections import namedtuple
+from slam.fast_slam2.Models import _EKF, Meas, LandmarkConstants
+from slam.robot_physics.RobotPhysics2D import RobotPhysics2D
+from slam.fast_slam2.FastSLAM2Particle import FastSLAM2Particle
+from typing import List
+
+class FastSLAM1Particle(FastSLAM2Particle):
+  '''
+  Class representing a particle in FastSLAM2
+  '''
+  def __init__(self, robot_physics: RobotPhysics2D, particle_id=None, **kwargs):
+    # random=None, map=None, num_landmarks_fixed=False, new_landmark_threshold=0.1, id=None
+    super().__init__(robot_physics, particle_id, **kwargs)
+
+  def _update_dependencies_on_landmark(self, landmark: _EKF,  pose_mean, pose_cov, meas):
+    est_meas_data = self._robot_physics.compute_meas_model(pose_mean, landmark.mean)
+    _, landmark.meas_jac_land = self._robot_physics.compute_meas_jacobians(pose_mean, landmark.mean)
+    landmark.innovation = meas.data - est_meas_data
+
+    landmark.Q = meas.cov + landmark.meas_jac_land @ landmark.cov @ landmark.meas_jac_land.T
+    landmark.Q_inv = np.linalg.inv(landmark.Q)
```

Comment:
Any risk of Q being singular?

Reply:
I do not believe so.

Reply:
Specifically, [this](https://stats.stackexchange.com/questions/395766/inverse-of-the-covariance-matrix-of-a-multivariate-normal-distribution) post suggests that a covariance matrix would be singular iff the variables have perfect correlation. However, Q is the covariance of a joint distribution, and it is very unlikely that its variables have perfect correlation. In fact, based on experiments so far, we haven't seen this use case. We will consider this further after this pull request.